timemachine/fe/free_energy.py

```diff
@@ -725,3 +722,185 @@ def compute_intermediate_result(lambdas: Sequence[float]) -> PairBarResult:
     boxes = [get_state(lamb).boxes for lamb in lambdas]
 
     return results, frames, boxes
+
+
+def run_sims_hrex(
+    initial_states: Sequence[InitialState],
+    md_params: MDParams,
+    n_frames_per_iter: int,
+    temperature: float,
+    n_swap_attempts_per_iter: Optional[int] = None,
+    verbose: bool = True,
+) -> Tuple[PairBarResult, List[StoredArrays], List[NDArray], HrexDiagnostics]:
+    r"""Sample from a sequence of states using nearest-neighbor Hamiltonian Replica EXchange (HREX).
```

Comment:
General question: does the algorithm in this PR still assume some sort of order dependence in the lambda schedule / set of initial states to be effective? (i.e. nearby neighbors have to have good overlap?). At first glance, since we rely on near-by-neighbor swaps (as opposed to global swaps from all possible permutations), if we say accidentally randomly shuffled the initial states, then HREX as currently implemented would fail? 

(This is not a blocker for the PR or anything, just wanted to double check)

Reply:
Added mention of this to the docstring in https://github.com/proteneer/timemachine/pull/1128/commits/814a8c825e353f63ddc9480b64e7e752e5f8f5c8